lucene/analysis/kuromoji/src/java/org/apache/lucene/analysis/ja/dict/BinaryDictionary.java

```diff
@@ -46,13 +50,31 @@
   public static final String POSDICT_HEADER = "kuromoji_dict_pos";
   public static final int VERSION = 1;
   
+  static final String CLASSPATH = "classpath:";
+  static final String FILE = "file:";
+
+  private final String resourcePrefix;
   private final ByteBuffer buffer;
   private final int[] targetMapOffsets, targetMap;
   private final String[] posDict;
   private final String[] inflTypeDict;
   private final String[] inflFormDict;
   
   protected BinaryDictionary() throws IOException {
+    this(CLASSPATH);
+  }
+
+  /**
+   * @param resourcePrefix - where to load resources (dictionaries) from. Either "file:", followed
+   * by a filesystem directory and optionally a filename-prefix, or "classpath:", followed by a
+   * classpath location prefix, or just plain "classpath:" to use this class's name as the prefix.
+   */
+  protected BinaryDictionary(String resourcePrefix) throws IOException {
```

Comment:
If the resourcePrefix parameter takes only "classpath:" or "file:", would it be possible to use enum here instead of arbitrary String (for safety)?.

Reply:
And just wondering, why do we need "file:" prefix? I feel like that loading the dictionary data from arbitrary location could be problematic... (Am I missed anything?)

Reply:
enum is a good idea; I'll update the patch. I added "file:" to enable unit testing. We can't load a test dictionary from the classpath, I think. Another possibility would be to pass in dictionaries directly somehow. EG we could maybe have {{...DictionaryBuilder}} {{FST.save()}} to an in-memory buffer, but this seemed like a more intrusive change since we can already save to a set of files.

Reply:
I missed the tests. OK, thanks!